### PR TITLE
Upgrade to Kafka 3.8.0

### PR DESCRIPTION
**Pull request by k-wall**
Changes to get compatibility with Kafka 3.8.0

### CODE REVIEW

File: kafka-server/src/test/java/com/ozangunalp/kafka/server/BrokerConfigTest.java [hunk 1]
```diff
@@ -8,168 +8,167 @@
...
-        assertThat(properties).containsEntry(KafkaConfig.BrokerIdProp(), "1");
-        assertThat(properties).containsEntry(KafkaConfig.QuorumVotersProp(), "1@:9094");
-        assertThat(properties).containsEntry(KafkaConfig.ListenersProp(), "BROKER://:9093,PLAINTEXT://:9092,CONTROLLER://:9094");
-        assertThat(properties).containsEntry(KafkaConfig.ProcessRolesProp(), "broker,controller");
-        assertThat(properties).containsEntry(KafkaConfig.ControllerListenerNamesProp(), "CONTROLLER");
-        assertThat(properties).containsEntry(KafkaConfig.InterBrokerListenerNameProp(), "BROKER");
-        assertThat(properties).containsEntry(KafkaConfig.AdvertisedListenersProp(), "PLAINTEXT://:9092,BROKER://:9093");
-        assertThat(properties).containsEntry(KafkaConfig.EarlyStartListenersProp(), "BROKER,CONTROLLER");
-        assertThat(properties).containsEntry(KafkaConfig.ListenerSecurityProtocolMapProp(), "BROKER:PLAINTEXT,CONTROLLER:PLAINTEXT,PLAINTEXT:PLAINTEXT");
+        assertThat(properties).containsEntry("broker.id", "1");
```
Comment (k-wall) [inline kafka-server/src/test/java/com/ozangunalp/kafka/server/BrokerConfigTest.java:17]: With constants now coming from so many different places, a mistake would be easy.  Testing against the string gives us confidence that the production code is achieving the right config.

**Comment by k-wall** [comment]
@robobario could you review this too?

**Comment by k-wall** [comment]
Apache Kafka RC3 has several binding votes already, so in preparation for the Apache Kafka 3.8.0 release, I've undrafted this PR.

**Comment by k-wall** [comment]
@ozangunalp if you have chance, could you review please?

**Comment by ozangunalp** [comment]
@k-wall yes i have it on my list. But I'll need a couple of days.

**Comment by ozangunalp** [comment]
It looks good to me. I pushed a small change to pull the remaining config names.

**Review by k-wall — COMMENTED** [review]
lgtm

**Comment by k-wall** [comment]
thanks, could you add an approval so I can merge this.

**Review by ozangunalp — APPROVED** [review]

**Pull request merged**